### ISSUE TITLE
Warning: /lib64/libc.so.6: version `GLIBC_2.14' not found

### ISSUE THREAD

**Issue by ilyazub** [node-gfx/node-canvas-prebuilt#44]
At my company we are using `node-canvas-prebuilt@~2.0.0-alpha.2` to build PNG sprite images. Everything works fine.

Suddenly, on some machines, `yarn` is reinstalling `node-canvas-prebuilt` even when dependencies remain unchanged. Don't know why - it's another question.

`yarn` downloads [canvas-prebuilt-v2.0.0-alpha.2-node-v57-linux-x64](https://github.com/node-gfx/node-canvas-prebuilt/releases/download/v2.0.0-alpha.2/canvas-prebuilt-v2.0.0-alpha.2-node-v57-linux-x64.tar.gz). This is a correct behavior. But after installation build script is failing with error.
```bash
Warning: /lib64/libc.so.6: version `GLIBC_2.14' not found (required by ./node_modules/canvas-prebuilt/canvas/build/Release/canvas.node) Use --force to continue.
# Absolute path was ommited.
```

I made a backup of correct `canvas.node` and we are replacing wrong one when issue occurs. After replacement everything works as usual without errors.
Re-installation of `node_modules` and `yarn cache clean` doesn't help.

It seems like `canvas.node` in that tarball was changed after release. At instance, file size differs from the previous one. I don't know if that possible, but this is my only assumption.
```bash
  File: './canvas-correct.node'
  Size: 276110    	Blocks: 552        IO Block: 4096   regular file
Device: fc00h/64512d	Inode: 2547939     Links: 1
Access: (0755/-rwxr-xr-x)  Uid: ( 1000/    ilya)   Gid: ( 1000/    ilya)
Access: 2018-06-13 15:06:32.979344840 +0300
Modify: 2017-11-04 21:46:12.000000000 +0200
Change: 2018-06-06 13:01:30.943530902 +0300
 Birth: -
  File: './canvas-wrong.node'
  Size: 273568    	Blocks: 536        IO Block: 4096   regular file
Device: fc00h/64512d	Inode: 2548115     Links: 1
Access: (0755/-rwxr-xr-x)  Uid: ( 1000/    ilya)   Gid: ( 1000/    ilya)
Access: 2018-06-13 15:17:57.862660000 +0300
Modify: 2018-05-02 18:02:22.000000000 +0300
Change: 2018-06-13 15:18:00.810626362 +0300
 Birth: -
```

Versions of `Node`, `npm` and `yarn` are the same on every development machine.
**Node.js version**: `v8.11.2`
**yarn version**: `1.3.2`
**npm version**: `5.6.0`

OS and libc versions are the same too: CentOS 6.9 and GNU libc v2.12.

To summarize, I have two questions:
1. Why `canvas.node` was changed in archive file of June 2017 release?
2. How to tell `yarn` to not to re-install installed binary (probably, this question should be asked somewhere else).

**Comment by ilyazub** [comment]
That's sad :disappointed:

Thanks for the answer.

**Comment by chearon** [comment]
Sorry about that. If I find the time maybe I can try building on a CentOS docker image instead of Debian, since it seems to have the oldest libc.

**Comment by ilyazub** [comment]
I'll try it too. Also, please publish git repository of `chearon/canvas-prebuilt` image, because for now I need to use something like [ImageLayers](https://imagelayers.io/?images=chearon%2Fcanvas-prebuilt:5) to inspect `Dockerfile`.

**Comment by chearon** [comment]
Cool, it's right [here](https://github.com/node-gfx/node-canvas-prebuilt/blob/master/docker/Dockerfile). Once you get the build deps, hopefully all the commands beyond that Just Work.
